dragome-js-jre/src/main/java/java/lang/Class.java

```diff
@@ -226,25 +226,51 @@ public T newInstance() throws InstantiationException, IllegalAccessException
 		return (T) ScriptHelper.eval("o", this);
 	}
 
+	private static boolean isInstanceInterfaceRecursive(final String clsBaseName, final Class cls) {
+		final Class<?>[] interfaces = cls.getInterfaces();
+		for(final Class<?> interfaceClass : interfaces)
+		{
+			if (interfaceClass.getName().equals(clsBaseName))
+			{
+				return true;
+			}
+			else if (isInstanceInterfaceRecursive(clsBaseName, interfaceClass))
+			{
+				return true;
+			}
+		}
+		return false;
+	}
+	
 	/**
 	 * Determines if the specified Object is assignment-compatible with the object represented by this Class.
+	 * https://docs.oracle.com/javase/8/docs/api/java/lang/Class.html#isInstance-java.lang.Object-
 	 */
 	public boolean isInstance(Object obj)
```

Comment:
And it would be nice to add a test for this reflection issue in com.dragome.tests.ReflectionAPITests suite.